src/engine/Settings.cs

```diff
@@ -197,10 +198,21 @@ public void ApplyGraphicsSettings()
     /// </summary>
     public void ResetToDefaults()
     {
-        // var defaults = new Settings();
+        var defaults = new Settings();
 
-        // TODO: apply the default values
-        throw new NotImplementedException();
+        Type type = GetType();
+        Type defaultsType = defaults.GetType();
+
+        PropertyInfo[] properties = type.GetProperties();
+        PropertyInfo[] defaultProperties = defaultsType.GetProperties();
+
+        for (int i = 0; i < properties.Length; ++i)
+        {
+            if (properties[i].PropertyType.IsPrimitive && properties[i].CanWrite)
```

Comment:
Why is `IsPrimitive` here? I think it would be better to use the JSON ignore properties if something needs to be ignored.